### PR TITLE
Multi-column assignment

### PR DESCRIPTION
**Pull request by rrjbca**
## Description
Add support for multi-column assignment from functions returning either 2d arrays or tuples of 1d arrays. New syntax applied to `mccl_galaxies.yml` where `schechter_lf` returns two arrays to be stored as two columns in the `red_galaxies` table named `redshift` and `magnitude`:

```
tables:
  red_galaxies:
    redshift, magnitude: !skypy.galaxy.schechter_lf
      redshift: $z_range
      M_star: $M_star_red
      phi_star: $phi_star_red
      alpha: $alpha_red
      m_lim: $mag_lim
      fsky: $fsky
```

## Checklist
- [x] Follow the [Contributor Guidelines](https://github.com/skypyproject/skypy/blob/master/CONTRIBUTING.md)
- [x] Write unit tests
- [x] Write documentation strings
- [x] Assign someone from your working team to review this pull request
- [x] Assign someone from the infrastructure team to review this pull request

### CODE REVIEW

**Comment by rrjbca** [comment]
I have used `/` as the character for splitting column names, but if there is a strong preference for using `,` or any other particular character then say so and we can consider changing it.

**Comment by ntessore** [comment]
I have very strong preference for normal tuple assignment syntax `redshift, magnitude`.

**Comment by ntessore** [comment]
But I absolutely love how far the configuration files have come with this.

**Comment by rrjbca** [comment]
> I have very strong preference for normal tuple assignment syntax `redshift, magnitude`.

Makes sense, I have made that change now

File: skypy/pipeline/_pipeline.py [hunk 1]
```diff
@@ -163,15 +178,22 @@ def execute(self):
...
+            if job in self.skip_jobs:
                 continue
             elif job in self.config:
                 settings = self.config.get(job)
                 setattr(self, job, self.get_value(settings))
             else:
                 table, column = job.split('.')
                 settings = self.table_config[table][column]
-                getattr(self, table)[column] = self.get_value(settings)
+                names = re.split(',\s?', column)
```
Comment (ntessore) [inline skypy/pipeline/_pipeline.py:189]: Allow arbitrary whitespace either side of the delimiter?

File: skypy/pipeline/tests/test_pipeline.py [hunk 2]
```diff
@@ -115,6 +122,34 @@ def test_pipeline():
...
+
+
+@pytest.mark.parametrize(('na', 'nt'), [(2, 3), (4, 3), (3, 2), (3, 4)])
+def test_multi_column_assignment_failure(na, nt):
+
+    # Test multi-column assignment failure with too few/many columns
+    config = {'tables': {
+                'multi_column_test_table': {
+                  'a/b/c': ('skypy.pipeline.tests.test_pipeline.multi_column_array', [7, na]),
+                  'd/e/f': ('skypy.pipeline.tests.test_pipeline.multi_column_tuple', [7, nt])}}}
```
Comment (ntessore) [inline skypy/pipeline/tests/test_pipeline.py:146]: uses slash delimiter

**Review by ntessore — CHANGES_REQUESTED** [review]

**Review by ntessore — APPROVED** [review]
🎉 🎉 🎉

**Pull request merged**